rpc/src/v1/impls/eth.rs

```diff
@@ -744,9 +743,14 @@ impl<C, SN: ?Sized, S: ?Sized, M, EM, T: StateInfo + 'static> Eth for EthClient<
 
 		// check if we're still syncing and return empty strings in that case
 		{
-			//TODO: check if initial sync is complete here
-			//let sync = self.sync;
-			if /*sync.status().state != SyncState::Idle ||*/ self.client.queue_info().total_queue_size() > MAX_QUEUE_SIZE_TO_MINE_ON {
+			let status = self.sync.status();
+			let client = &self.client;
+			let warping = match self.snapshot.status() {
```

Comment:
I don't think you need to check for this, I think `is_major_importing` should already return `true` if we're warping (although would be nice if you could test this).